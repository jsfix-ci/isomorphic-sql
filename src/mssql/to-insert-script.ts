
export function toInsertScript(table: string, values: any[]) {
    const columns = Object.keys(values[0]).sort();
    return `
INSERT INTO ${table}
(${columns.join(', ')})
VALUES
${getValues(columns, values)}`;
}

function getValues(columns, values) {
    return values.map(v => {
        return columns.map(c => {
            return `'${v[c]}'`
        }).join(', ')
    }).map(exp => `(${exp})`).join(', ')
}