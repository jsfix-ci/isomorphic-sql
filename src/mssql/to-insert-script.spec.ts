import {toInsertScript} from './to-insert-script';
describe('asValue', () => {
    it('returns a parenthesized value expression', () => {
        expect(toInsertScript('foods', [
            {name: 'hotdogs'},
            {name: 'chimichangas'},
        ])).toMatchSnapshot();
    });
});