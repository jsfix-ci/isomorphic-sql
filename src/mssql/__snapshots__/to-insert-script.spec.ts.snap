// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asValue returns a parenthesized value expression 1`] = `
"
INSERT INTO foods
(name)
VALUES
('hotdogs'), ('chimichangas')"
`;
